src/Compilers/VisualBasic/Portable/Semantics/TypeInference/TypeArgumentInference.vb

```diff
@@ -586,7 +586,8 @@ HandleAsAGeneralExpression:
                             ParameterType,
                             Parameter,
                             MatchGenericArgumentToParameter.MatchBaseOfGenericArgumentToParameter,
-                            inferenceRestrictions)
+                            inferenceRestrictions,
+                            addedHints:=Nothing)
```

Comment:
>addedHints [](start = 28, length = 10)

I didn't find where `addedHints` is used. What is its purpose in the context of this change?

Reply:
It is used on line 1659 in this file (https://github.com/dotnet/roslyn/pull/22295/files#diff-6e2be44971b9721836d1f298f24ccae7R1659).

Reply:
What is its purpose in the context of this change? e.g. how it is used to more correctly implement the language feature, if it fixes a bug, and what is the symptom that is fixed? It isn't obvious what the purpose of the change on line 1659 is.

Reply:
@gafter The purpose of the logic on line 1659 is to continue with "indirect" inference when  direct inference doesn't run into any error conditions and doesn't result in any new hints.

Reply:
A number of tuple tests were affected by this as well as non-tuple scenario (see ShapeMismatchInOneArgument unit-test in TypeArgumentInference.vb). Before, the indirect inference was performed only when InferTypeArgumentsFromArgumentDirectly was returning false.